src/pages/NewConversation/DirectMessageForm/useDirectMessageForm.js

```diff
@@ -0,0 +1,128 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { createSelector } from 'reselect';
+
+// Actions
+import { sendMessageRequest } from '../../../actions/chats-actions';
+import {
+  removeAllReceivers,
+  createNewConversation,
+} from '../../../actions/newconversation-actions';
+
+// Routes
+import { CONVERSATION_WITHOUT_ID } from '../../../consts/routes';
+
+// Utils
+import createChatHeadingString from '../../../utils/createheadingChatString';
+import pickAvatarToDisplay from '../../../utils/pickAvatarToDisplay';
+
+// Selector which return found conversation or undefined
+const conversationExistingLocallyDMSelector = createSelector(
+  [
+    (state) => state.newconversation.receivers,
+    (state) => state.auth.user._id,
+    (state) => state.main.chats,
+  ],
+  (receivers, userId, chats) => {
+    if (receivers.length === 0) {
+      return false;
+    }
+    const membersIds = [
+      ...new Set([...receivers.map((receiver) => receiver.id), userId]),
```

Comment:
Why set here converted to array,  are you trying to commit duplicates in array that way?